polaris.shopify.com/src/pages/api/icons/v0/[icon].tsx

```diff
@@ -0,0 +1,26 @@
+import type {NextApiResponse, NextApiRequest} from 'next';
+import fs from 'fs';
+
+const handler = async (req: NextApiRequest, res: NextApiResponse) => {
+  try {
+    const {icon} = req.query;
+    const iconsvg = fs.readFileSync(
+      `../polaris-icons/dist/svg/${icon}`,
+      'utf-8',
+    );
```

Comment:
Is the entire monorepo deployed with the site? Wonder, if we should resolve the svgs from the `node_modules`? e.g. Something like:
```js
    const iconSvg = await fs.promises.readFile(
      path.join(
        require.resolve('@shopify/polaris-tokens'),
        `../dist/svg/${icon}`
      ),
      'utf-8',
    );
```

Reply:
It is deployed!

Reply:
Oh Interesting! I still think it might be safer to pull from the `node_modules` so we're accessing assets from published versions and not whatever is on `polaris/main`. That said, I'm not very familiar with Polaris site infra or if my suggestion would even by pass the yarn workspace symlinks so I'll trust you two!

Reply:
I don't think this is the best approach but we couldn't access the files with imports. We should try the require resolve method ^